server/cpho/rules.py

```diff
@@ -0,0 +1,105 @@
+from server.rules_framework import add_rule, auto_rule
+
+
+@auto_rule
+def is_admin(user):
+    return user.username == "admin"
+
+
+@auto_rule
+def is_hso(user):
+    return user.username == "hso"
+
+
+@auto_rule
+def is_program(user):
+    return user.username == "program"
+
+
+@auto_rule
+def is_branch_lead(user):
+    return True
+
+
+@auto_rule
+def is_branch_user(user):
+    return True
+
+
+@auto_rule
+def is_branch_lead_or_user(user):
+    return is_branch_user(user) or is_branch_lead(user)
+
+
+@auto_rule
+def is_indicator_lead(user, indicator):
+    return True
+
+
+@auto_rule
+def indicator_of_users_branch(user, indicator):
+    return True
+
+
+@auto_rule
+def can_edit_indicator(user, indicator):
+    return (
+        is_admin(user)
+        or is_hso(user)
+        or (is_program(user) and is_indicator_lead(user, indicator))
+        or (
+            is_program(user)
+            and is_branch_lead(user)
+            and indicator_of_users_branch(user, indicator)
+        )
+    )
+
```

Comment:
Rules can be combined using bool operators, this can be re-written as 

```python
rules.add_rule("can_edit_indicator", (
  is_admin | 
  is_hso | 
  (is_program & is_indicator_lead) | 
  ( is_program & is_branch_lead & indicator_of_users_branch )
)
```
It's a question of style, though, it won't always work if you need to pass a different variable, but it should work in this case because the rules library will ensure to only pass the indicator argument to rules who take 2 arguments

Reply:
After discussion with Alex, decided to keep this as the syntax is probably easier to understand